bnb-chain/akula
# parlia: support fast finality hard fork;

this pr will partially support BSC fast finality, Akula could full sync after the hard fork, it contains:
1. add boneh and lynn hard fork logic;
2. add BLS key manager to verify sig;
3. parse and verify fast finality block header;
4. add some precompile contracts;
5. support newly reward and slash logic;
6. add fast finality vote pool, support vote new head, verify and collect votes;
7. prepare new block attestation;
